src/components/post/post.service.ts

```diff
@@ -67,34 +76,53 @@ export class PostService {
     return await Promise.all(posts.map((dto) => this.secure(dto, session)));
   }
 
+  async loadPostable(resource: CommentableRef): Promise<Postable> {
+    const parentNode = isIdLike(resource)
+      ? await this.repo.getBaseNode(resource, Resource)
+      : resource;
+    if (!parentNode) {
+      throw new NotFoundException('Resource does not exist', 'resourceId');
+    }
+    const parent = isBaseNode(parentNode)
+      ? await this.resources.loadByBaseNode(parentNode)
+      : parentNode;
+
+    try {
+      await this.resourcesHost.verifyImplements(parent.__typename, Postable);
+    } catch (e) {
+      throw new NonPostableType(e.message);
+    }
+    return parent as Postable;
+  }
+
   private async secure(
     dto: UnsecuredDto<Post>,
     session: Session,
   ): Promise<Post> {
-    const securedProps = await this.authorizationService.secureProperties(
-      Post,
-      dto,
-      session,
-    );
+    return this.privileges.for(session, Post).secure(dto);
+  }
 
-    return {
-      ...dto,
-      ...securedProps,
-      canDelete: await this.repo.checkDeletePermission(dto.id, session),
-    };
+  async getPermissionsFromResource(resource: CommentableRef, session: Session) {
+    const parent = await this.loadPostable(resource);
+    const parentType = await this.resourcesHost.getByName(
+      parent.__typename as 'Postable',
+    );
+    return this.privileges.for(session, parentType, parent).forEdge('posts');
   }
 
   async update(input: UpdatePost, session: Session): Promise<Post> {
     const object = await this.readOne(input.id, session);
 
     const changes = this.repo.getActualChanges(object, input);
+    this.privileges.for(session, Post, object).verifyChanges(changes);
```

Comment:
@CarsonF - Mind feedback on this specifically? App seems to be working as expected, but I am like 75% confident in this line and then where I also call the privileges in `delete` (line 125)

Reply:
Actually... should this be replaced by the `getPermissionsFromResource`?

Reply:
> Actually... should this be replaced by the `getPermissionsFromResource`?

Yes

Reply:
Just a heads up. Fixed delete. Update need a little more love tonight or tomorrow.